p2p/trust/metric_test.go

```diff
@@ -61,30 +61,32 @@ func TestTrustMetricStopPause(t *testing.T) {
 	tm := NewMetricWithConfig(config)
 
 	// Allow some time intervals to pass and pause
-	time.Sleep(50 * time.Millisecond)
+	tm.NextTimeInterval()
+	tm.NextTimeInterval()
 	tm.Pause()
-	// Give the pause some time to take place
-	time.Sleep(10 * time.Millisecond)
 
 	first := tm.Copy().numIntervals
 	// Allow more time to pass and check the intervals are unchanged
-	time.Sleep(50 * time.Millisecond)
-	assert.Equal(t, first, tm.numIntervals)
+	tm.WaitForTimeIntervalToPass()
```

Comment:
why we can't call `tm.NextTimeInterval()` here too similar to above?

Reply:
the wait method will wait until the ticker in the go-routine fires again, which is what the test requires

Reply:
ok. I will take a look why ticker is so important here